### PR TITLE
treewide: stdenv.is -> stdenv.hostPlatform.is

### PR DESCRIPTION
**Pull request by JohnRTitor**
Commands ran:
```
fd --type f "\.nix" | xargs sd --fixed-strings "stdenv.is" "stdenv.hostPlatform.is"
fd --type f "\.nix" | xargs sd --fixed-strings "stdenv'.is" "stdenv'.hostPlatform.is"
fd --type f "\.nix" | xargs sd --fixed-strings "clangStdenv.is" "clangStdenv.hostPlatform.is"
fd --type f "\.nix" | xargs sd --fixed-strings "gccStdenv.is" "gccStdenv.hostPlatform.is"
fd --type f "\.nix" | xargs sd --fixed-strings "stdenvNoCC.is" "stdenvNoCC.hostPlatform.is"
fd --type f "\.nix" | xargs sd --fixed-strings "inherit (stdenv) is" "inherit (stdenv.hostPlatform) is"
fd --type f "\.nix" | xargs sd  "buildStdenv.is" "buildStdenv.hostPlatform.is"
fd --type f "\.nix" | xargs sd  "effectiveStdenv.is" "effectiveStdenv.hostPlatform.is"
fd --type f "\.nix" | xargs sd  "originalStdenv.is" "originalStdenv.hostPlatform.is"
```

Following https://github.com/NixOS/nixpkgs/pull/341407

~`stdenv.isLinux` -> `stdenv.hostPlatform.isLinux`, is pretty much easy expansion of alias. This should cause no rebuilds.~

~However some packages have stdenv.isLinux in `doCheck`, meaning package tests have to be run when building in linux. So instead of `hostPlatform` there use `buildPlatform` there. This could cause some rebuilds.~

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ar/arti/package.nix [hunk 1]
```diff
@@ -28,7 +28,7 @@ rustPlatform.buildRustPackage rec {
 
   buildInputs =
     [ sqlite ]
-    ++ lib.optionals stdenv.isLinux [ openssl ]
```
Comment (Artturin) [inline pkgs/by-name/ar/arti/package.nix:31]: These sneaked back in https://github.com/NixOS/nixpkgs/commit/5136e375f2e764f7d11ec87455ca05498f723af8 lol

**Comment by Artturin** [comment]
Just do the commands in https://github.com/NixOS/nixpkgs/pull/341407 instead of only isLinux

**Comment by emilazy** [comment]
I am unsure about the second commit here. Is it going to work with cross? Don’t we want a `canExecute` thing instead?

**Comment by Artturin** [comment]
> I am unsure about the second commit here. Is it going to work with cross? Don’t we want a `canExecute` thing instead?

https://github.com/NixOS/nixpkgs/blob/cac8a72e28e533916f63e529133f6efd3ed6a4d6/pkgs/stdenv/generic/make-derivation.nix#L249 canExecute is added by stdenv on top

However not sure if it's added when doing overrideAttrs  like done in lua-modules

**Comment by JohnRTitor** [comment]
> I am unsure about the second commit here. Is it going to work with cross?

It should work I suppose, the checks needs to be enabled on the build machine, not on the host machine where it would run.

**Comment by emilazy** [comment]
Formatter is unhappy.

**Comment by JohnRTitor** [comment]
Not related to this PR, the errors are occuring as the original files are not formatted. We are only touching a line, formatting them here is out of scope

**Comment by emilazy** [comment]
CI only complains if files that were formatted before the PR become unformatted, so I don’t think that’s true.

**Comment by jopejoe1** [comment]
It complains about `pkgs/applications/radio/sdrangel/default.nix`, `pkgs/by-name/er/erlang-language-platform/package.nix` and `pkgs/by-name/im/imhex/package.nix` not being formated.

**Comment by JohnRTitor** [comment]
Yeah, it seems like nixfmt will throw errors if line char limits have been exceeded and use an alternate style.

Done, this should make the CI happy, I have still decided to make the commit seperately though, for easier skimming through the diff. The PR can be squash merged if needed.

**Review by JohnRTitor — COMMENTED** [review]
Eval successfully passed but a merge conflict is shown. Intend to merge after rebasing, as while we race for the eval to complete again likely another conflict will arise

**Pull request merged**